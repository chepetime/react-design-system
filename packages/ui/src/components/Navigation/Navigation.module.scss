@use "./../../styles" as styles;

:root {
  --navigation-background-color: #{styles.$pink-500};
}

:global .dark {
  --navigation-background-color: #{styles.$indigo-500};
}

.Navigation {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--navigation-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
