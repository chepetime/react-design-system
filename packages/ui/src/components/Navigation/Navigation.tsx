import React from "react";
import classnames from "classnames";
import css from "./Navigation.module.scss";

export interface NavigationProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Navigation = React.forwardRef<HTMLDivElement, NavigationProps>(
  (props: NavigationProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Navigation)} ref={ref}>
        <p>Navigation component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Navigation.displayName = "Navigation";
