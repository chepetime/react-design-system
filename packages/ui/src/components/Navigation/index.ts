export { Navigation } from "./Navigation";
export type { NavigationProps } from "./Navigation";