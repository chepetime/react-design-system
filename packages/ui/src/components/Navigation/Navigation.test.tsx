
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { Navigation } from "./Navigation";
import { render, screen } from "@testing-library/react";

describe("Navigation", () => {
  it("renders without errors", () => {
    render(<Navigation />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<Navigation>Some children</Navigation>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<Navigation>Some children</Navigation>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<Navigation />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
