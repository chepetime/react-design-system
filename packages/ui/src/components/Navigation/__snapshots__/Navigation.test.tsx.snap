// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot 1`] = `
<div>
  <div
    class="Navigation"
  >
    <p>
      Navigation component working!
    </p>
    <div />
  </div>
</div>
`;
